app/api/profiles/[id]/publish/route.ts

```diff
@@ -0,0 +1,18 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { updateUserData } from '@/backend/profile/profile.service'
+import { authorizeUser } from '@/lib/auth'
+
+export async function POST(request: NextRequest, id: string) {
+  try {
+    await authorizeUser()
+
+    const updatedUser = await updateUserData(id, { isPublished: true })
```

Comment:
**nitpick:** This function should be named `updateUserProfileById`. Currently it suggests that we operate on `User` domain instead of `Profile`